azkaban-common/src/main/java/azkaban/jobtype/JobTypePluginSet.java

```diff
@@ -37,8 +41,15 @@
   private final Map<String, Props> pluginPrivatePropsMap;
   private final Map<String, JobPropsProcessor> pluginJobPropsProcessor;
   private final Map<String, String> jobToClassName;
+  // Mapping for jobType to defaultProxyUser
+  private final Map<String, String> jobToDefaultProxyUser;
   private final Map<String, URL[]> jobToClassLoaderURLs;
 
+  // List of jobType classes which are allowed for the defaultProxyUsers feature
+  private final List<String> defaultProxyUsersJobTypeClasses;
+  // List of users which are not allowed as defaultProxyUsers
+  private final List<String> defaultProxyUsersFilter;
```

Comment:
the name is a bit confusing, perhaps, defaultProxyUserDenyList? This will also mean changes everywhere it is used, provided you keep it list.
More importantly, List is not appropriate data structure, since it is meant to filter out certain users, in such case a Set is more appropriate, then perhaps, the name could be kept as it is.

Reply:
Make sense. I will utilize set everywhere in this PR where List is a bad data structure.